### PR TITLE
ARROW-6218: [Java] Add UINT type test in integration to avoid potential overflow

### PR DESCRIPTION
**Pull request by tianchen92**
Related to [ARROW-6218](https://issues.apache.org/jira/browse/ARROW-6218).
As per discussion https://github.com/apache/arrow/pull/5002

For UINT type, when write/read json data in integration test, it extend data type(i.e. Long->BigInteger, Int->Long) to avoid potential overflow.

Like UINT8 the write side and read side code like this:
>case UINT8:
  generator.writeNumber(UInt8Vector.getNoOverflow(buffer, index));
  break;
 
>BigInteger value = parser.getBigIntegerValue();
buf.writeLong(value.longValue());

Should add a test to avoid potential overflow in the data transfer process.

### CODE REVIEW

**Comment by tianchen92** [comment]
cc @praveenbingo

File: java/vector/src/test/java/org/apache/arrow/vector/ipc/TestJSONFile.java [hunk 1]
```diff
@@ -401,4 +406,39 @@ public void testWriteReadMapJSON() throws IOException {
...
     }
   }
+
+  @Test
+  public void testNoOverFlowWithUINT() {
+    try (final UInt8Vector uInt8Vector = new UInt8Vector("uint8", allocator);
+        final UInt4Vector uInt4Vector = new UInt4Vector("uint4", allocator);
+        final UInt1Vector uInt1Vector = new UInt1Vector("uint1", allocator)) {
+
+      long[] longValues = new long[]{0x8000000000000000L, 0x7fffffffffffffffL, 0xffffffffffffffffL};
```
Comment (emkornfield) [inline java/vector/src/test/java/org/apache/arrow/vector/ipc/TestJSONFile.java]: Long.MIN_VALUE, Long.MAX_VALUE, -1L?

Comment (tianchen92) [reply java/vector/src/test/java/org/apache/arrow/vector/ipc/TestJSONFile.java]: Right, fixed :)

**Review by praveenbingo — APPROVED** [review]
+1 LGTM. thanks @tianchen92
